src/WebService.cpp

```diff
@@ -53,19 +53,32 @@ namespace weatherserver {
 
     web::http::client::http_client client(uriBuilder.to_string());
 
-    return client.request(web::http::methods::GET)
-      .then([](web::http::http_response requestResponse)
+    auto allLocations = client.request(web::http::methods::GET)
+      .then([this, &countryName, limit](web::http::http_response requestResponse)
         {
           std::cout << "fetchAllLocations() request completed!" << std::endl;
 
           return requestResponse.extract_json();
         })
-      .then([](web::json::value jsonValue)
+      .then([this, &countryName, limit](web::json::value jsonValue)
         {
           std::cout << "JSON extracted from fetchAllLocations() completed!" << std::endl;
-          auto results = jsonValue.at(U("results"));
-          return results;
-        });
+
+          auto metaData = jsonValue.at(U("meta"));
+          size_t foundResultsNumber = metaData.at(U("found")).as_integer();
+
+          web::json::value result;
+
+          if (foundResultsNumber <= limit) {
+            result = jsonValue.at(U("results"));
+          }
+          else {
+            result = this->fetchAllLocations(countryName, foundResultsNumber);
```

Comment:
Its ok for quick fix to use recursion here. But in general it will be better to have a separate synchronous function for obtaining json from constructed url.